src/utils.c

```diff
@@ -59,13 +63,17 @@ GthreeObject *
 load_model (const char *name)
 {
   GError *error = NULL;
+  const char *path;
   g_autoptr(GFile) base = NULL;
   g_autoptr(GFile) file = NULL;
   g_autoptr(GFile) parent = NULL;
   g_autoptr(GBytes) bytes = NULL;
   g_autoptr(GthreeLoader) loader = NULL;
 
-  base = g_file_new_for_path ("../models");
+  path = g_getenv ("GNOME_HEXGL_DATADIR");
+  if (path == NULL)
+    path = DATADIR;
+  base = g_file_new_for_path (g_strjoin("/", path, "models", NULL));
```

Comment:
This leaks the return value of strjoin. Anyway you probably want datadir = ..new_for_path (path); base = g_file_get_child (datadir, "models").

Reply:
Done, not sure if base will leake something as I'm reusing it